pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,461 @@
+    // type is ExternalName.
+    // More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+    // +optional
+    string clusterIP = 3;
+
+    // externalIPs is a list of IP addresses for which nodes in the cluster
+    // will also accept traffic for this service.  These IPs are not managed by
+    // Kubernetes.  The user is responsible for ensuring that traffic arrives
+    // at a node with this IP.  A common example is external load-balancers
+    // that are not part of the Kubernetes system.
+    // +optional
+    repeated string externalIPs = 5;
+
+    // Supports "ClientIP" and "None". Used to maintain session affinity.
+    // Enable client IP based session affinity.
+    // Must be ClientIP or None.
+    // Defaults to None.
+    // More info: https://kubernetes.io/docs/concepts/services-networking/service/#virtual-ips-and-service-proxies
+    // +optional
+    string sessionAffinity = 7;
+
+    // LoadBalancer will get created with the IP specified in this field.
+    // This feature depends on whether the underlying cloud-provider supports specifying
+    // the loadBalancerIP when a load balancer is created.
+    // This field will be ignored if the cloud-provider does not support the feature.
+    // +optional
+    string loadBalancerIP = 8;
+
+    // If specified and supported by the platform, this will restrict traffic through the cloud-provider
+    // load-balancer will be restricted to the specified client IPs. This field will be ignored if the
+    // cloud-provider does not support the feature."
+    // More info: https://kubernetes.io/docs/tasks/access-application-cluster/configure-cloud-provider-firewall/
+    // +optional
+    repeated string loadBalancerSourceRanges = 9;
+
+    // externalName is the external reference that kubedns or equivalent will
+    // return as a CNAME record for this service. No proxying will be involved.
+    // Must be a valid RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+    // and requires Type to be ExternalName.
+    // +optional
+    string externalName = 10;
+
+    // externalTrafficPolicy denotes if this Service desires to route external
+    // traffic to node-local or cluster-wide endpoints. "Local" preserves the
+    // client source IP and avoids a second hop for LoadBalancer and Nodeport
+    // type services, but risks potentially imbalanced traffic spreading.
+    // "Cluster" obscures the client source IP and may cause a second hop to
+    // another node, but should have good overall load-spreading.
+    // +optional
+    string externalTrafficPolicy = 11;
+}
+
+// OperatorConfig defines a config used to install an external operator.
+message OperatorSpec {
+    // manifest_path is the URL for the operator install manifest.
+    string manifest_path = 1;
+
+    // namespace is the namespace the operator manifest and CR are installed into.
+    string namespace = 2;
```

Comment:
isn't it more natural to install in the same namespace as the CR?

Reply:
As which CR? This is for both the operator and its CR so I think this is already doing what you're suggesting?

Reply:
Could you please explain: this is for both the operator and its CR?  I suspect I'm missing a critical detail in how this is intended to be used.

Reply:
yes, it's for both. what this will do in practice is install the raw external operator CR (spec below) into the cluster, get the external operator yaml from the URL and apply that into the cluster. both will be applied to the given namespace. 
after that, the operator ignores this struct (does nor reconcile).